pandas/core/indexes/extension.py

```diff
@@ -170,6 +172,29 @@ class ExtensionIndex(Index):
     __le__ = _make_wrapped_comparison_op("__le__")
     __ge__ = _make_wrapped_comparison_op("__ge__")
 
+    def __getitem__(self, key):
+        result = self._data[key]
+        if isinstance(result, type(self._data)):
+            return type(self)(result, name=self.name)
+
+        # Includes cases where we get a 2D ndarray back for MPL compat
+        return result
+
+    def __iter__(self):
+        return self._data.__iter__()
+
+    @property
+    def _ndarray_values(self) -> np.ndarray:
+        return self._data._ndarray_values
+
+    def dropna(self, how="any"):
+        if how not in ("any", "all"):
+            raise ValueError(f"invalid how option: {how}")
+
+        if self.hasnans:
+            return self._shallow_copy(self._data[~self._isnan])
+        return self._shallow_copy()
```

Comment:
Why are you overwriting the base Index one? 

Also, this dropped the docstring.

Reply:
Because the base class uses ._values, where we want ._data here

Reply:
But `_values` and `_data` is the same?

Reply:
I guess.  Past-me must have thought it not-obvious that this would always hold.  If it can be removed, go for it.

Reply:
Did you also see my docstring comment?

Reply:
I did.  In this case I think removing the method makes sense.  More generally I wonder if we can use a metaclass or something to automatically inherit docstrings and remove a lot of boilerplate (cc @bashtage IIRC you do something like this in arch)

Reply:
Maybe open a new issue to see if we can do this smarter? 

But for 1.0.0, I would just add back the docstring